ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx

```diff
@@ -324,14 +326,17 @@ const TopologyComponent = ({
                             controller.getGraph().reset();
                             controller.getGraph().layout();
                         },
-                        legendCallback: () => {
-                            // console.log('hi');
-                        },
                     })}
                 />
             }
         >
             <VisualizationSurface state={{ selectedIds }} />
+            <Popover
+                aria-label="Network graph legend"
+                bodyContent={<LegendContent />}
+                hasAutoWidth
+                reference={() => document.getElementById('legend') as HTMLButtonElement}
```

Comment:
this is following an example on the popover patternfly docs (https://www.patternfly.org/v4/components/popover#popover-selector-ref)